src/lib.rs

```diff
@@ -202,12 +215,21 @@ pub async fn get_intensities(
             let start_date = start_date.format("%Y-%m-%dT%H:%MZ");
             let end_date = end_date.format("%Y-%m-%dT%H:%MZ");
 
-            let url = format!("{BASE_URL}/regional/intensity/{start_date}/{end_date}/{path}");
+            if *target != Target::National {
+                let url = format!("{BASE_URL}/regional/intensity/{start_date}/{end_date}/{path}");
+
+                tokio::spawn(async move {
```

Comment:
Why do we spawn a tokio task just for one request here and in the `else` branch?

Seems unnecessary "complicated" 🤔

Reply:
duplicated the code because we are calling different methods 
 `get_intensities_for_url_national` vs `get_intensities_for_url`

Reply:
oh sorry, I see what you mean now, for some reason I didn't realise we were inside the loop that spawn the tokio tasks for when we get multiple date ranges' data 🤦